src/providers/wms/qgsxyzsourcewidget.cpp

```diff
@@ -71,8 +71,6 @@ void QgsXyzSourceWidget::setSourceUri( const QString &uri )
   mAuthSettings->setConfigId( mSourceParts.value( QStringLiteral( "authcfg" ) ).toString() );
 
   setInterpretation( mSourceParts.value( QStringLiteral( "interpretation" ) ).toString() );
-
-  mIsValid = true;
```

Comment:
shouldn't this be

    validate();

instead? If the uri string is empty then the widget won't be valid, and should emit the validChanged() signal accordingly

Reply:
Validation already happens above in the line 55. When `mEditUrl->setText()` is called this automatically triggers `validate` and all signals are emitted.

This explicit assignment then overrides validation results and does not any signal.

Reply:
Ok, fair enough!